controller/api/destination/server_test.go

```diff
@@ -81,38 +84,19 @@ func TestGet(t *testing.T) {
 		}
 	})
 
-	t.Run("Returns endpoints", func(t *testing.T) {
-		server := makeServer(t)
-		defer server.clusterStore.UnregisterGauges()
-
-		stream := &bufferingGetStream{
-			updates:          make(chan *pb.Update, 50),
-			MockServerStream: util.NewMockServerStream(),
-		}
-		defer stream.Cancel()
-		errs := make(chan error)
-
-		// server.Get blocks until the grpc stream is complete so we call it
-		// in a goroutine and watch stream.updates for updates.
-		go func() {
-			err := server.Get(&pb.GetDestination{Scheme: "k8s", Path: fmt.Sprintf("%s:%d", fullyQualifiedName, port)}, stream)
-			if err != nil {
-				errs <- err
-			}
-		}()
+	t.Run("Returns endpoints (IPv4)", func(t *testing.T) {
+		addr := fmt.Sprintf("%s:%d", podIP1, port)
+		testReturnEndpoints(t, fullyQualifiedName, addr)
+	})
 
-		select {
-		case update := <-stream.updates:
-			if updateAddAddress(t, update)[0] != fmt.Sprintf("%s:%d", podIP1, port) {
-				t.Fatalf("Expected %s but got %s", fmt.Sprintf("%s:%d", podIP1, port), updateAddAddress(t, update)[0])
-			}
+	t.Run("Returns endpoints (IPv6)", func(t *testing.T) {
+		addr := fmt.Sprintf("[%s]:%d", podIP1v6, port)
+		testReturnEndpoints(t, fullyQualifiedNameIPv6, addr)
+	})
 
-			if len(stream.updates) != 0 {
-				t.Fatalf("Expected 1 update but got %d: %v", 1+len(stream.updates), stream.updates)
-			}
-		case err := <-errs:
-			t.Fatalf("Got error: %s", err)
-		}
+	t.Run("Returns endpoints (dual-stack)", func(t *testing.T) {
+		addr := fmt.Sprintf("[%s]:%d", podIPv6Dual, port)
+		testReturnEndpoints(t, fullyQualifiedNameDual, addr)
 	})
```

Comment:
Can we add some state change test cases? E.g. when there's an IPV4 address and then an IPV6 address is added to supersede it and when there's both address families and then the IPV6 address is removed and we fall back to IPV4.

Reply:
Yep, definitely.